### ISSUE TITLE
linux 64bit version

### ISSUE THREAD

**Issue by benzkji** [pingdynasty/OwlControl#10]
I cannot run OWLNest on Xubuntu 64bit. 

    ./OwlControl: error while loading shared libraries: libX11.so.6: cannot open shared object file: No such file or directory

this is probably because it's a 32bit build, right?

**Comment by giuliomoro** [comment]
can you do `ldd OwlControl` ?

**Comment by benzkji** [comment]
yes. do I need to install some dependencies? I guess it's not that trivial, as I read, having 32bit support on Linux...?

    --- ~/Apps » ldd OwlControl
	linux-gate.so.1 =>  (0xf77a8000)
	libX11.so.6 => not found
	libXext.so.6 => not found
	libasound.so.2 => not found
	libdl.so.2 => /lib32/libdl.so.2 (0xf7782000)
	libfreetype.so.6 => not found
	libpthread.so.0 => /lib32/libpthread.so.0 (0xf7763000)
	librt.so.1 => /lib32/librt.so.1 (0xf775a000)
	libstdc++.so.6 => not found
	libm.so.6 => /lib32/libm.so.6 (0xf7705000)
	libgcc_s.so.1 => /usr/lib32/libgcc_s.so.1 (0xf76e7000)
	libc.so.6 => /lib32/libc.so.6 (0xf752f000)
	/lib/ld-linux.so.2 (0x5656c000)

**Comment by giuliomoro** [comment]
I just built it on Ubuntu 16 64 bit.

You can use the binary below or try to to rebuild it for your system. Building it is actually very easy and the best solution.
You would clone the repo,  then
```
$ cd OwlControl/Builds/Linux && make
```
if the `make` fails it is probably because of some missing includes (and associated library), for instance you get a message like:
```
in file included from ../../JuceLibraryCode/juce_gui_basics.cpp:9:0:
../../JuceLibraryCode/modules/juce_gui_basics/juce_gui_basics.cpp:126:35: fatal error: X11/Xcursor/Xcursor.h: No such file or directory
   #include <X11/Xcursor/Xcursor.h>
```

To fix these messages (there may be a few of those) do:
find what package contains the missing file
```
$ apt-file search X11/Xcursor/Xcursor.h
```
which returns 
```
libxcursor-dev: /usr/include/X11/Xcursor/Xcursor.h
```
the file is contained in `libxcursor-dev`, so install this:
```
sudo apt-get install libxcursor-dev
```

and then start over from `make` and fix the errors as they show up. It does not take long really.

In my case i had to install these, so you may want to start with this command and see if you get any other missing dependencies:
```
sudo apt-get install libxcursor-dev libxinerama-dev libxrandr-dev libfreetype6-dev
```

The compiled file, in case you need it:
[OwlControl.zip](https://github.com/pingdynasty/OwlControl/files/1091814/OwlControl.zip)

**Comment by pingdynasty** [comment]
@benzkji did the binary work for you?

**Comment by giuliomoro** [comment]
@benzkji you will at least need the non-dev versions of those libraries to run it.

**Comment by benzkji** [comment]
ldd of the 64bit version is all ok, and at first glance everything works fine! thx alot!

**Comment by benzkji** [comment]
one thing though, it doesn't show up as a "window" in my desktop env (alt-tab switcher, etc). means I can only switch to OwlControl by minimizing other windows. is this normal? It could be due to my xubuntu configuration, dont know?

**Comment by benzkji** [comment]
also, "load patch from" file doesnt show anything, and blocks the app (cannot move the window anymore, must kill OwlControl).

Strange issues, as it seems to run for most people on linux? I'm on xubuntu/xfce, if this might help. Also, it isn't that important, I'll try to compile with OwlProgramm anyway.

Leaving this open, if you want to investigate, but feel free to close it yourself ;-)

**Comment by pingdynasty** [comment]
Seems there's something wrong with the window management, modal pop-ups should work fine.
I'm using LXDE with no probs, but on 32bit Ubuntu 14.
Might be worth upgrading the Juce library version and rebuild. Are you using any other Juce apps on this platform?

**Comment by benzkji** [comment]
I've downloaded and run OwlNest, until I saw I got Firmwar v12 ;-) Could this have some bad side effects? Do you mean I could rebuild OwlControl on my machine, and check if it works?

**Comment by benzkji** [comment]
I'v managed to compile it. Needed to use gcc-5 ([Howto on Ubuntu](https://gist.github.com/beci/2a2091f282042ed20cda), seems a Jucer version problem, [see here](https://github.com/falkTX/Carla/issues/395) and [here](https://github.com/open-ephys/plugin-GUI/issues/92)). As you already pointed out, upgrading jucer could help. I dont have other Jucer apps (as far as I know ;-)

The open dialog problem persists, with the self built version.

**Comment by benzkji** [comment]
I've tried to build it, because your version dumped a

    JUCE v4.2.2
    [1]    14932 illegal hardware instruction (core dumped)  ~/Apps/OwlControl

(MacBook Pro, ubuntu zesty)
